rr_computer_vision/src/endline_detection.cpp

```diff
@@ -0,0 +1,142 @@
+/** @file endline_detection.cpp
+ *  @brief Magenta endline detection
+ *
+ * Topics Subscribed:
+ *   /zed/rgb/image_rect_color
+ *
+ * Service called:
+ *   /Supervisor/count_lap
+ *
+ * @author Waleed Ahmed (w29ahmed)
+ * @author Yuchi(Allan) Zhao
+*/
+
+#include "endline_detection.hpp"
+#include "opencv2/highgui/highgui.hpp"
+#include "opencv2/imgproc/imgproc.hpp"
+#include <vector>
+
+// Constructor
+EndlineDetection::EndlineDetection(ros::NodeHandle nh) : it_(nh_)
+{
+  detection_status = false;
+  endline_counter = 0;
+  client_ = nh_.serviceClient<std_srvs::Trigger>("/Supervisor/count_lap");
+  
+  test_subscriber = it_.subscribe("/zed/rgb/image_rect_color", 1, &EndlineDetection::ImgCb, this);
```

Comment:
Function name is image callback but what is this callback processing? Make the name clearer so someone doesnt have to read the inside to figure out what it does